.github/workflows/ci.yaml

```diff
@@ -728,6 +728,7 @@ jobs:
       (github.event_name != 'push' || github.event.repository.full_name == 'home-assistant/core')
       && github.event.inputs.lint-only != 'true'
       && (needs.changes.outputs.test_full_suite == 'true' || needs.changes.outputs.tests_glob)
+      && (needs.pip-check.result != 'failure')
```

Comment:
This doesn't do anything. `needs -> pip-check` means `pip-check` is required to be successful before the `pytest` job is run. I.e. if `pip-check` is skipped, pytest would run at all.

Reply:
As per your link: `If a job fails, all jobs that need it are skipped unless the jobs use a conditional expression that causes the job to continue.`
So the if statement overrides the "success" requirement